contracts/src/systems/npc/contracts.cairo

```diff
@@ -1,49 +1,63 @@
 #[dojo::contract]
 mod npc_systems {
-    use starknet::{ContractAddress, get_caller_address};
+    use starknet::{get_caller_address};
     use starknet::info::BlockInfo;
 
     use eternum::constants::NPC_CONFIG_ID;
     use eternum::models::realm::{Realm, RealmTrait};
-    use eternum::models::npc::{
-        Npc, pack_characs, unpack_characs, random_role, random_sex, Characteristics,
-        random_characteristics
-    };
+    use eternum::models::npc::{Npc, pack_characs, random_characteristics};
     use eternum::models::last_spawned::{LastSpawned, ShouldSpawnImpl};
     use eternum::systems::npc::utils::assert_ownership;
     use eternum::models::config::NpcConfig;
     use eternum::systems::npc::interface::INpc;
 
-    use debug::PrintTrait;
     use core::Into;
 
+    #[derive(Drop, starknet::Event)]
+    struct NpcSpawned {
+        #[key]
+        realm_id: u128,
+        npc_id: u128,
+    }
+
     #[event]
     #[derive(Drop, starknet::Event)]
-    enum Event {}
+    enum Event {
+        NpcSpawned: NpcSpawned,
+    }
 
     #[generate_trait]
     impl InternalFunctions of InternalFunctionsTrait {
         // trigger the mood changes based on when the user clicks on the harvest weat/fish
         fn change_characteristics(
             self: @ContractState,
             world: IWorldDispatcher,
-            realm_id: felt252,
-            npc_id: felt252,
+            realm_id: u128,
+            npc_id: u128,
             characteristics: felt252
         ) {
             assert_ownership(world, realm_id);
 
             let caller_address = get_caller_address();
-            let old_npc = get!(world, (npc_id), (Npc));
+            let old_npc = get!(world, npc_id, (Npc));
             // otherwise seeing this error on compilation
             // let __set_macro_value__ = Npc { entity_id: npc_id, realm_id, mood, role: old_role.sex, sex: old_sex.role};
-            set!(world, (Npc { entity_id: npc_id, realm_id, characteristics }));
+            set!(
+                world,
+                (Npc {
+                    entity_id: npc_id,
+                    realm_id,
+                    characteristics,
+                    character_trait: old_npc.character_trait,
+                    name: old_npc.name
+                })
+            );
         }
     }
 
     #[external(v0)]
     impl NpcImpl of INpc<ContractState> {
-        fn spawn_npc(self: @ContractState, world: IWorldDispatcher, realm_id: felt252) -> felt252 {
+        fn spawn_npc(self: @ContractState, world: IWorldDispatcher, realm_id: u128, character_trait: felt252, name: felt252) -> u128 {
```

Comment:
Same comment as above, should we pass sex + role just like the trait, as we would get them from the prompt ?

Reply:
Changed